### DISCUSSION TITLE
`expose` should use xlib composite extension

### DISCUSSION THREAD

**Discussion by ralt** [stumpwm/stumpwm#1108] [General]
This extension was made exactly for this purpose: take pixels of a window and render them, instead of making the window itself smaller.

Fortunately, CLX implements bindings for it https://github.com/sharplispers/clx/blob/master/extensions/composite.lisp

Unfortunately, that extension is not included in its ASD file in the version hosted on quicklisp. It was only enabled in this commit: https://github.com/sharplispers/clx/commit/18efe6ae9fde7fa644ac7c64cda724b7deadec0b

(The documentation is here: https://www.x.org/releases/X11R7.5/doc/compositeproto/compositeproto.txt)

**Comment by dmb2** [comment]
I have no objection to this, but I will not be devoting any dev time to it.  The expose features were ported on request from a ratpoison extension.

**Comment by equwal** [comment]
> Unfortunately, that extension is not included in its ASD file in the version hosted on quicklisp. It was only enabled in this commit: [sharplispers/clx@18efe6a](https://github.com/sharplispers/clx/commit/18efe6ae9fde7fa644ac7c64cda724b7deadec0b)

This fact has become false; it is in the quicklisp distro now.

**Comment by equwal** [comment]
It is not just `expose` that would benefit from this. Lots of redrawing operations can be sped up, and flickering can be reduced. `expose` uses a bunch of calls to `vsplit` and `hsplit`, so it is just where such things are more noticeable.

For example, try inserting a short input item with lots of completions, then holding down tab. The box will flash and change very slowly because it is using the hardware device.
